tests/mpi/periodic_neighbor_01.cc

```diff
@@ -0,0 +1,371 @@
+ *       -------------------------------------------------
+ *       |           |     |  |  |     |     |           |
+ *       |           |     |-----|     |     |           |
+ *       |           |     |  |  |     |     |           |
+ *       |           -------------------------           |
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       -------------------------------------------------
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       -------------------------------------------------
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       -------------------------------------------------
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       |           -------------------------           |
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       -------------------------------------------------
+ *                    Bottom periodic boundary
+ *
+ *
+ * =====================
+ * [ Passing criteria: ]
+ * =====================
+ *
+ * The test will be passed if the main three assertions are passed.
+ */
+
+#include "../tests.h"
+#include <deal.II/distributed/grid_refinement.h>
+#include <deal.II/distributed/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_boundary_lib.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/lac/vector.h>
+
+#include <fstream>
+#include <string>
+
+
+using namespace dealii;
+
+template <int dim>
+struct periodicity_tests
+{
+  typedef TriaIterator<CellAccessor<dim>> cell_iterator;
+  typedef typename Triangulation<dim>::active_cell_iterator active_cell_iterator;
+  periodicity_tests();
+  ~periodicity_tests();
```

Comment:
You implement an empty destructor below. Since the class is non-virtual, you might just as well skip the destructor entirely. The compiler will do everything for you.